ecosystem/scenarios/utils.py

```
@@ -0,0 +1,260 @@
+            install_requires.add(pkg_name)
+
+        # Collect all extras
+        extras_require = set()
+        for extra_reqs in requirements.get("extras_require", {}).values():
+            for req in extra_reqs:
+                pkg_name = req.split(">=")[0].split("==")[0].split("<")[0].split(">")[0].split("[")[0].strip()
+                extras_require.add(pkg_name)
+
+        return {"install_requires": install_requires, "extras_require": extras_require}
+
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)